src/pytorch_lightning/strategies/fully_sharded.py

```diff
@@ -132,6 +142,25 @@ def process_group(self) -> Any:
             self._process_group = torch.distributed.new_group()
         return self._process_group
 
+    def lightning_module_state_dict(self) -> Dict[str, Any]:
+        """Returns model state."""
+        assert self.model is not None
+        return self.model.state_dict()
+
+    def connect(self, model: "pl.LightningModule") -> None:
+        """Called by the accelerator to connect the accelerator and the model with this plugin."""
+        # TODO: Wait for this issue to resolve and remove this blocker
+        # https://github.com/facebookresearch/fairscale/issues/648
+        # Also make sure to update the tests
+        if not is_overridden("configure_sharded_model", self.lightning_module) and len(list(model.parameters())) == 0:
```

Comment:
because of https://github.com/facebookresearch/fairscale/issues/648
this is a blocker, but the user can still reinitialize the model and CPU, which will work. Added tests for both cases.